src/loaders/IVolumeLoader.ts

```diff
@@ -65,3 +64,60 @@ export interface IVolumeLoader {
   // TODO explicitly passing a `LoadSpec` is now rarely useful. Remove?
   loadVolumeData(volume: Volume, loadSpec?: LoadSpec, onChannelLoaded?: PerChannelCallback): void;
 }
+
+/** Abstract class which allows loaders to accept and return types that are easier to transfer to/from a worker. */
+export abstract class ThreadableVolumeLoader implements IVolumeLoader {
+  /** Unchanged from `IVolumeLoader`. See that interface for details. */
+  abstract loadDims(loadSpec: LoadSpec): Promise<VolumeDims[]>;
+
+  /**
+   * Creates an `ImageInfo` object from a `LoadSpec`, which may be passed to the `Volume` constructor to create an
+   * empty volume that can accept data loaded with the given `LoadSpec`.
+   *
+   * Also returns a new `LoadSpec` that may have been modified from the input `LoadSpec` to reflect the constraints or
+   * abilities of the loader. This new `LoadSpec` should be used when constructing the `Volume`, _not_ the original.
+   */
+  abstract createImageInfo(loadSpec: LoadSpec): Promise<[ImageInfo, LoadSpec]>;
```

Comment:
What do you think of this?  Because I saw it used in a couple of places.

```suggestion
  abstract createImageInfo(loadSpec: LoadSpec): Promise<{imageInfo:ImageInfo, loadSpec:LoadSpec}>;
```